helix-tui/src/buffer.rs

```diff
@@ -510,9 +510,13 @@ impl Buffer {
     pub fn clear_with(&mut self, area: Rect, style: Style) {
         for x in area.left()..area.right() {
             for y in area.top()..area.bottom() {
-                let cell = &mut self[(x, y)];
-                cell.reset();
-                cell.set_style(style);
+                if let Some(cell) = self.get_mut(x, y) {
+                    cell.reset();
+                    cell.set_style(style);
+                } else {
+                    // Skip this render if the window size has changed
+                    continue;
+                }
```

Comment:
I'm a bit wary about changing this: the window size won't change halfway through a render. If `clear_with` is called with an out of bounds rectangle then that's a bug that should be addressed and skipping it here would hide the bug.

Reply:
I had a problem with that. Why window can not change it size during render?

Reply:
Thank you for the detailed answer. `clear_with` is not related with crashes I had indeed.